crates/jstz_node/openapi.json

```diff
@@ -422,26 +422,46 @@
       "Content": {
         "oneOf": [
           {
-            "type": "object",
-            "required": [
-              "DeployFunction"
-            ],
-            "properties": {
-              "DeployFunction": {
+            "allOf": [
```

Comment:
Code changes look fine to me, but the generated spec seems a bit weird, for instance
```json
      "Content": {
        "oneOf": [
          {
            "allOf": [
              {
                "$ref": "#/components/schemas/DeployFunction"
              },
              {
                "type": "object",
                "required": [
                  "#type"
                ],
                "properties": {
                  "#type": {
                    "type": "string",
                    "enum": [
                      "DeployFunction"
                    ]
                  }
                }
              }
            ]
          },
          ...
```
ideally we want just the first element of `allOf` without `allOf` itself. Is this doable?

Reply:
The correct way to add discriminator in OAS is the `discriminator` field, however, utoipa only generates this if the enum is marked as `serde(untagged)` which then generates json like:
```
{
  "Ok": {
    "account": "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx",
    "updated_balance": 20
  }
}
```
Instead of:
```
{
  "Ok": {
    "_type": "Deposit",
    "account": "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx",
    "updated_balance": 20
  }
}
```
(Note: I changed `#type` to `_type`)

The `_type` will end up being a field on ts structs and is used to discriminate between variants in match statements.

Reply:
I get your point and that's fine. I see that now [ReceiptResult](https://github.com/jstz-dev/jstz/pull/663/files#diff-6a082d1d5ac19606f82468d82595327a0ea053178edbaeee2e6121b5e1f036d1R84) has this `content = "inner"` thing. Should we apply this to all enums? Then we can get rid of this `allOf` thing here so that it looks like the `ReceiptResult` [schema](https://github.com/jstz-dev/jstz/pull/663/files#diff-ed955db8c0de308210b3093969021b0e41c7cd2af81a8ca3c665e1e69fe89141R788).

Reply:
No, I think its rather unconventional to apply this format for types that are not meant to be container types

Reply:
I think tag + content is useful for container types but we shouldn't overuse it as it becomes a burden in the generated code.

Reply:
leaving this unresolved temporarily for https://github.com/jstz-dev/jstz/pull/644#pullrequestreview-2442122051